src/Microsoft.TestPlatform.ObjectModel/Utilities/XmlRunSettingsUtilities.cs

```diff
@@ -264,6 +264,73 @@ public static DataCollectionRunSettings GetDataCollectionRunSettings(string runS
         }
 
         /// <summary>
+        /// Get InProc DataCollection Run settings
+        /// </summary>
+        /// <param name="runSettingsXml">
+        /// The run Settings Xml.
+        /// </param>
+        /// <returns>Data collection run settings.</returns>
+        [SuppressMessage("Microsoft.Security.Xml", "CA3053:UseXmlSecureResolver",
+            Justification = "XmlReaderSettings.XmlResolver is not available in core. Suppress until fxcop issue is fixed.")]
+        public static DataCollectionRunSettings GetInProcDataCollectionRunSettings(string runSettingsXml)
+        {
+            // use XmlReader to avoid loading of the plugins in client code (mainly from VS).
+            if (!string.IsNullOrWhiteSpace(runSettingsXml))
+            {
+                runSettingsXml = runSettingsXml.Trim();
+                using (StringReader stringReader1 = new StringReader(runSettingsXml))
+                {
+                    XmlReader reader = XmlReader.Create(stringReader1, ReaderSettings);
+
+                    // read to the fist child
+                    XmlReaderUtilities.ReadToRootNode(reader);
+                    reader.ReadToNextElement();
+
+                    // Read till we reach In Proc IDC element or reach EOF
+                    while (!string.Equals(reader.Name, Constants.InProcDataCollectionRunSettingsName)
+                           &&
+                           !reader.EOF)
+                    {
+                        reader.SkipToNextElement();
+                    }
+
+                    // If reached EOF => IDC element not there
+                    if (reader.EOF)
+                    {
+                        return null;
+                    }
+
+                    // Reached here => In Proc IDC element present.
+                    return DataCollectionRunSettings.FromXml(reader, Constants.InProcDataCollectionRunSettingsName, Constants.InProcDataCollectorsSettingName, Constants.InProcDataCollectorSettingName);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Get ExecutionThreadApartmentStateKey from runsettings.
+        /// </summary>
+        /// <param name="settingsXml"> The runsettings. </param>
+        /// <returns> Thread Apartment State. </returns>
+        public static PlatformApartmentState GetExecutionThreadApartmentState(string settingsXml)
```

Comment:
Added ExecutionThreadApartmentState prop to RunConfiguration. Removed this method. Not combining the data collector settings reading because not using then always together in other places.